### PR TITLE
bank storage inside dynamic map

### PR DESCRIPTION
**Pull request by jpcima**
This is the first attempt to reciprocate the bank storage method in libOPNMIDI.
There is no public API for the moment.

### CODE REVIEW

**Comment by Wohlstand** [comment]
I'll check this thing a bit later, especially I'll try various XG songs to validate multi-bank and instruments management...

File: src/opnmidi_midiplay.cpp [hunk 1]
```diff
@@ -1376,8 +1372,7 @@ void OPNMIDIplay::NoteUpdate(uint16_t MidCh,
     const int16_t tone    = info.tone;
     const uint8_t vol     = info.vol;
     const size_t midiins = info.midiins;
-    const size_t insmeta = info.insmeta;
```
Comment (Wohlstand) [inline src/opnmidi_midiplay.cpp]: Removal of this breaks active_note.cpp unit test!
Can you fix it?

**Comment by jpcima** [comment]
Updated the unit test. There's a thing though: I make it use the empty instrument, because OPNMIDI does not have the embedded banks.
Also, I converted the build system to build catch main one time only.

**Comment by Wohlstand** [comment]
> Updated the unit test. There's a thing though: I make it use the empty instrument, because OPNMIDI does not have the embedded banks.

Good! :smile: 
I'll review this at evening again when I'll come to my home...

> Also, I converted the build system to build catch main one time only.

Neat! :fox_face: :+1:

**Pull request merged**